app/core/SnapKeyring/SnapKeyring.ts

```diff
@@ -0,0 +1,115 @@
+ * - `removeAccount`: Initiates the process of removing an account with user confirmation and handling the user input.
+ * - `redirectUser`: Redirects the user to a specified URL with a message to complete signing. This method is used to handle asynchronous signing requests.
+ * - `addressExists`: Returns a boolean indicating if an address exists in the keyring.
+ */
+export const snapKeyringBuilder = (
+  controllerMessenger: SnapKeyringBuilderMessenger,
+  getSnapController: () => SnapController,
+  persistKeyringHelper: () => Promise<void>,
+  removeAccountHelper: (address: string) => Promise<unknown>,
+): { (): SnapKeyring; type: string } => {
+  const builder = () => new SnapKeyring(getSnapController(), {
+    addressExists: async (address) =>
+      (
+        await controllerMessenger.call('KeyringController:getAccounts')
+      ).includes(address.toLowerCase()),
+
+    redirectUser: async (snapId: string, url: string, message: string) => {
+      Logger.log(
+        `SnapKeyring: redirectUser called with \n
+        - snapId: ${snapId} \n
+        - url: ${url} \n
+        - message: ${message} \n`,
+      );
+    },
+
+    saveState: async () => {
+      await persistKeyringHelper();
+    },
+
+    addAccount: async (
+      address: string,
+      snapId: string,
+      handleUserInput: (accepted: boolean) => Promise<void>,
+      accountNameSuggestion = '',
+      displayConfirmation = false,
+    ) => {
+      Logger.log(
+        `SnapKeyring: addAccount called with \n
+        - address: ${address} \n
+        - handleUserInput: ${handleUserInput} \n
+        - snapId: ${snapId} \n
+        - accountNameSuggestion: ${accountNameSuggestion} \n
+        - displayConfirmation: ${displayConfirmation}`,
+      );
+
+      //approve everything for now because we have not implemented snap account confirmations yet
+      await handleUserInput(true);
+      await persistKeyringHelper();
+      const account = controllerMessenger.call(
+        'AccountsController:getAccountByAddress',
+        address,
+      );
+      if (!account) {
+        throw new Error(
+          `Internal account not found for address: ${address}`,
+        );
+      }
+
+      // Set the selected account to the new account
+      controllerMessenger.call(
+        'AccountsController:setSelectedAccount',
+        account.id,
+      );
+    },
+
+    removeAccount: async (
+      address: string,
+      snapId: string,
+      handleUserInput: (accepted: boolean) => Promise<void>,
+    ) => {
+      Logger.log(
+        `SnapKeyring: removeAccount called with \n
+        - address: ${address} \n
+        - handleUserInput: ${handleUserInput} \n
+        - snapId: ${snapId} \n`,
+      );
+      // approve everything for now because we have not implemented snap account confirmations yet
+      await removeAccountHelper(address);
+      await handleUserInput(true);
```

Comment:
Just re-ordering things, since the comment is related to `handleUserInput` and `persistKeyringHelper`:

```suggestion
      await removeAccountHelper(address);

      // Approve everything for now because we have not implemented snap account confirmations yet
      await handleUserInput(true);
```

Reply:
I based this order off of what is already [in production on the extension](https://github.com/MetaMask/metamask-extension/blob/ed2850844e2af8381284468313c920b5c6e5084a/app/scripts/lib/snap-keyring/snap-keyring.ts#L365-L367). do you think we should still change the order?